src/BenchmarksServer/Git.cs

```diff
@@ -27,12 +29,34 @@ public static string Clone(string path, string repository, string branch = null)
 
         public static void Checkout(string path, string branchOrCommit)
         {
-            RunGitCommand(path, $"checkout {branchOrCommit}");
+            RunGitCommand(path, $"checkout {branchOrCommit}", retries: 5);
+        }
+
+        private static ProcessResult RunGitCommand(string path, string command, bool throwOnError = true, int retries = 0)
+        {
+            return RetryOnException(retries, () => ProcessUtil.Run("git", command, workingDirectory: path, throwOnError: throwOnError, timeout: CommandTimeout));
```

Comment:
Maybe this timeout should be one level up as each command might require a different timeout. Clone could be the only one to use it.